qiita_db/test/test_metadata_template.py

```diff
@@ -1363,6 +1364,38 @@ def test_create_bad_sample_names(self):
             PrepTemplate.create(self.metadata, self.new_raw_data,
                                 self.test_study, self.data_type)
 
+    def test_create_unknown_sample_names(self):
+        # set two real and one fake sample name
+        self.metadata_dict['NOTREAL'] = self.metadata_dict['SKB7.640196']
+        del self.metadata_dict['SKB7.640196']
+        self.metadata = pd.DataFrame.from_dict(self.metadata_dict,
+                                               orient='index')
+        # Test error raised and correct error given
+        # Try/Except used so can test the text of the error raised
+        try:
```

Comment:
The problem with this approach is that if no exception is raised this test could still potentially pass, you need to assertRaises (with a context manager) and then check for the appropriate values.